spec/models/task_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe Task, type: :model do
+  it { should validate_presence_of(:project).with_message(/can't be blank or invalid/) }
+  it { should validate_presence_of :name }
+  it { should validate_presence_of :state }
+  it { should belong_to(:project) }
+
+  let (:task) { build(:task) }
+
+  it "should be in the default state 'todo" do
+    expect(task.state).to eq("todo")
+  end
+
+  it "should be in the state 'in-progress'" do
+    task.state = :'in-progress'
+    expect(task.state).to eq("in-progress")
+  end
+
+  it "should be in the state 'done'" do
+    task.state = :done
+    expect(task.state).to eq("done")
+  end
```

Comment:
So I wouldn't say these verifications of setting the state are necessary. Though I can see how you would want to verify the states that could be set and would agree with that.
So I think instead a spec around the class method `states` would be more appropriate that just shows that the possible states are what you would expect them to be. I would probably even just check `Task.states.keys` as we don't really usually care about what the numeric values are.

If you did want to instead verify it the way you have here, I would still do it differently. Your spec basically says "Task should be in the state 'in-progress'" and "Task should be in the state 'done'" and in addition you already have "Task should be in the default state 'todo'".

The first one about the default state is fine, though you should prefer not to use words like "should" in a spec as it's implied. Something like "starts in the default state 'todo'".

The later ones would be best described using context blocks. So something like:
```
context "when the state is set to 'in-progress'" do
  before { task.state = :'in-progress' }
  it "has the state 'in-progress'" do
    expect(task.state).to eq("in-progress")
  end
end
```

So when you set it up like that with proper context so the RSpec reads more like a specification, it stands out even more that maybe you aren't testing what you want to be testing because of course setting a property to a value would give it that value. Bringing you back to maybe wanting to just verify the possible states and trust that the Rails enum system will ensure that you can only set `state` to one of the valid states.